tsconfig.json

```diff
@@ -12,7 +12,7 @@
     "module": "ESNext",
     "moduleDetection": "force",
     "allowJs": true,
-    "noEmit": false,
+    "noEmit": true,
```

Comment:
We don't actually compile with `tsc`, just typecheck. `vite` runs our compilation step via `esbuild`, so we can save some time/duplication by just telling `tsc` not to emit files.